libdebug/debugger/internal_debugger.py

```diff
@@ -993,30 +1026,104 @@ def get_thread_by_id(self: InternalDebugger, thread_id: int) -> ThreadContext:
 
         return None
 
-    def resolve_address(self: InternalDebugger, address: int) -> int:
+    def resolve_address(self: InternalDebugger, address: int, backing_file: str) -> int:
         """Normalizes and validates the specified address.
 
         Args:
             address (int): The address to normalize and validate.
+            backing_file (str): The backing file to resolve the address in.
 
         Returns:
             int: The normalized and validated address.
+
+        Raises:
+            ValueError: If the substring `backing_file` is present in multiple backing files.
         """
         maps = self.debugging_interface.maps()
-        return normalize_and_validate_address(address, maps)
 
-    def resolve_symbol(self: InternalDebugger, symbol: str) -> int:
+        if backing_file in ["default", "absolute"]:
+            if check_absolute_address(address, maps):
+                # If the address is absolute, we can return it directly
+                return address
+            elif backing_file == "absolute":
+                # The address is explicitly an absolute address but we did not find it
+                raise ValueError(
+                    "The specified absolute address does not exist. Check the address or specify a backing file.",
+                )
+            else:
+                # If the address was not found and the backing file is not "absolute",
+                # we have to assume it is in the main map
+                backing_file = self._get_process_full_path()
+                liblog.debugger(
+                    f"No backing file specified and no correspondant absolute address for {hex(address)}. Assuming {backing_file}."
+                )
+        elif (
+            backing_file == (full_backing_path := self._get_process_full_path())
+            or backing_file == "binary"
+            or backing_file == self._get_process_name()
+        ):
+            backing_file = full_backing_path
+
+        filtered_maps = []
+        unique_files = set()
+
+        for vmap in maps:
+            if backing_file in vmap.backing_file:
+                filtered_maps.append(vmap)
+                unique_files.add(vmap.backing_file)
+
+        if len(unique_files) > 1:
+            raise ValueError(
+                f"The substring {backing_file} is present in multiple, different backing files. The address resolution cannot be accurate."
+            )
+
+        if not filtered_maps:
+            raise ValueError(f"The specified string {backing_file} does not correspond to any backing file.")
```

Comment:
And here

Reply:
67a9609f8c813654d939a28ea5970f65eff64650